ext/oauth-client/Civi/Api4/Action/OAuthSysToken/Refresh.php

```diff
@@ -50,7 +50,7 @@ protected function getSelect() {
   }
 
   protected function doTask($row) {
-    if ($this->threshold >= 0 && \CRM_Utils_Time::getTimeRaw() < $row['expires'] - $this->threshold) {
+    if (!empty($token['access_token']) && ($this->threshold >= 0 && \CRM_Utils_Time::time() < $row['expires'] - $this->threshold)) {
```

Comment:
You must have meant `$row['access_token']`.

It runs ok in various scenarios with $row, but would still be looking for a second opinion as to what it's supposed to do if it's empty, and I'm wondering how it helps troubleshooting.

Reply:
I've removed it from the PR :-)